contracts/ReputationMiningCycle.sol

```diff
@@ -50,77 +79,110 @@ contract ReputationMiningCycle {
     _;
   }
 
-  function ReputationMiningCycle() public {
-    colonyNetworkAddress = msg.sender;
-    reputationMiningWindowOpenTimestamp = now;
-  }
-
-  function respondToChallenge(uint256 round, uint256 idx) public {
-    // TODO: Check challenge response is valid, relating to current challenge
-    // TODO: Check challenge response relates to this hash
-
-    // Assuming that the challenge response is correct...
-    uint256 opponentIdx = (idx % 2 == 1 ? idx-1 : idx + 1);
-
-    disputeRounds[round][idx].lastResponseTimestamp = now;
-    disputeRounds[round][idx].challengeStepCompleted += 1;
-    // If our opponent responded to this challenge before we did, we should
-    // reset their 'last response' time to now, as they aren't able to respond
-    // to the next challenge before they know what it is!
-    if (disputeRounds[round][idx].challengeStepCompleted == disputeRounds[round][opponentIdx].challengeStepCompleted) {
-      disputeRounds[round][opponentIdx].lastResponseTimestamp = now;
-    }
+  modifier challengeOpen(uint256 round, uint256 idx) {
+    // TODO: More checks that this is an appropriate time to respondToChallenge
+    require(disputeRounds[round][idx].lowerBound == disputeRounds[round][idx].upperBound);
+    _;
   }
 
-  function submitNewHash(bytes32 newHash, uint256 nNodes, uint256 entry) public {
-    //Check the ticket is an eligible one for them to claim
+  modifier entryQualifies(bytes32 newHash, uint256 nNodes, uint256 entry) {
+    // Check the ticket is an eligible one for them to claim
+    // TODO: Require minimum stake, that is (much) more than the cost required to defend the valid submission.
     require(entry <= IColonyNetwork(colonyNetworkAddress).getStakedBalance(msg.sender) / 10**15);
```

Comment:
This line is checking the minimum stake required, maybe we don't need the above todo?

Reply:
Is this just a matter of recalculating the minimum stake then? We should know this cost by now, right?